lib/menubar/utils.lua

```diff
@@ -345,6 +345,9 @@ function utils.parse_desktop_file(file)
         if program.Terminal == true then
             cmdline = utils.terminal .. ' -e ' .. cmdline
         end
+        if program.Path then
+            cmdline = 'sh -c "cd ' .. program.Path .. ' && ' .. cmdline .. '"'
```

Comment:
This wraps `cmdline` in quotes, but doesn't escape any quotes inside `cmdline`. This will very likely break for any `program.Exec` that contains quotes (single or double) itself.
You'll have to alter `cmdline` so that all quotes are escaped, including once that already had been escaped.

Reply:
and also this works only if path is exists, while spec not define the behavior, i'd rather use `;` instead of `&&`

Reply:
I think that if `Path` is specified, the expected behaviour would be to either run with CWD set to that directory or not at all.
Starting with any other directory, because we ignore the error from `cd` could lead to confusing, hard to debug behaviour.

Reply:
can I use something like:
`cmdline = 'cd ' .. program.Path .. ' && ' .. cmdline`
and modify [`lib/menubar/init.lua#L211`](https://github.com/awesomeWM/awesome/blob/master/lib/menubar/init.lua#L211) with
`awful.spawn.with_shell`
This may solve the quotes problem

Reply:
add new property for menuitem "with_shell", and set it here and and for program.Terminal

and at line 211 check if menuitem.with_shell is set and only when use spawn_with_shell